src/main/java/org/candlepin/resource/CloudRegistrationResource.java

```diff
@@ -83,4 +155,124 @@ private CloudRegistrationData getCloudRegistrationData(CloudRegistrationDTO clou
         registrationData.setSignature(cloudRegistrationDTO.getSignature());
         return registrationData;
     }
+
+    private CloudAuthenticationResultDTO processAdapterAuthResult(Principal principal,
+        CloudAuthenticationResult authResult) {
+        validateCloudAuthenticationResult(authResult);
+
+        String cloudInstanceId = authResult.getCloudInstanceId();
+        AnonymousCloudConsumer existingAnonConsumer = anonymousCloudConsumerCurator
+            .getByCloudInstanceId(cloudInstanceId);
+
+        String ownerKey = authResult.getOwnerKey();
+
+        // verify that the owner exists upstream and is entitled
+        boolean createAnonConsumer = false;
+        if (ownerKey == null || ownerKey.isBlank() || !authResult.isEntitled()) {
+            CloudAccountOrgSetupJobConfig jobConfig = CloudAccountOrgSetupJob.createJobConfig()
+                .setCloudAccountId(authResult.getCloudAccountId())
+                .setCloudOfferingId(authResult.getOfferId())
+                .setCloudProvider(authResult.getCloudProvider());
+
+            try {
+                this.jobManager.queueJob(jobConfig);
+            }
+            catch (JobException e) {
+                String errMsg = this.i18n.tr("An unexpected exception occurred " +
+                    "while scheduling job \"{0}\"", jobConfig.getJobKey());
+                log.error(errMsg, e);
+                throw new IseException(errMsg, e);
+            }
+
+            if (existingAnonConsumer == null) {
+                createAnonConsumer = true;
+            }
+
+            log.info(
+                "Cloud account org setup job created for account {} {} using offer {}",
+                authResult.getCloudProvider().shortName(), authResult.getCloudAccountId(),
+                authResult.getOfferId());
+        }
+
+        // Impl note: It is possible the owner and subscription exists through the adapters, but
+        // the owner and pool do not yet exist in Candlepin. If that is the case we will want to create
+        // an anonymous cloud consumer record and provide an anonymous registration token.
+        String anonymousConsumerUuid = existingAnonConsumer == null ? null : existingAnonConsumer.getUuid();
+        boolean ownerAndPoolExists = poolCurator.hasPoolForProduct(ownerKey, authResult.getProductId());
+        if ((!ownerAndPoolExists && existingAnonConsumer == null) || createAnonConsumer) {
```

Comment:
Updated. 

There was a problem with that algorithm not covering the case where a VM is authenticating again and has already created an anonymous consumer record and the pools and owner are not yet synced to Candlepin. This results in an anonymous consumer not needing to be created (`createAnonConsumer=false`) and we need to return an anonymous token, but since the token type is dictated by the `createAnonConsumer`, we return a standard token type.

I moved the token and token type to be based on `anonymousConsumerUuid` again.

This test case is covered by the spec test:
`CloudRegistrationSpecTest.shouldReceiveSameAnonConsumerUuidWhenReAuthenticatingForV2Auth`